lib/validator.js

```diff
@@ -158,58 +158,47 @@ function checkSubSchema(schema, parent, state) {
         if (typeof schema.additionalItems === 'boolean') {
         }
         else if (typeof schema.additionalItems === 'object') {
-            schema.additionalItems.should.not.be.an.Array();
+            schema.additionalItems.should.not.be.instanceOf(Array, 'additionalItems should not be an array, it shoud be a boolean or schema');
         }
         else should.fail(false,true,'additionalItems must be a boolean or schema');
     }
     if (schema.additionalProperties) {
         if (typeof schema.additionalProperties === 'boolean') {
         }
         else if (typeof schema.additionalProperties === 'object') {
-            schema.additionalProperties.should.not.be.an.Array();
+            schema.additionalProperties.should.not.be.instanceOf(Array, 'additionalProperties should not be an array, it should be a boolean or schema');
         }
         else should.fail(false,true,'additionalProperties must be a boolean or schema');
     }
     if (schema.maxItems) {
-        schema.maxItems.should.be.a.Number();
+        schema.maxItems.should.be.type(Number, 'maxItems should be a number');
```

Comment:
Is this correct? Looking at https://shouldjs.github.io/#assertion-type shouldn't the type (1st parameter) be a string?

Reply:
Several occurrences of the last issue.